### PR TITLE
[optim][BE] split test file into logical parts: SWA, LR, optim

### PR DESCRIPTION
**Pull request by janeyx99**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #101103
* __->__ #101100
* #101004

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @albanD

### CODE REVIEW

File: torch/autograd/gradcheck.py [hunk 1]
```diff
@@ -1427,8 +1427,6 @@ def _fast_gradcheck(func, func_out, inputs, outputs, eps, rtol,
 # If https://mypy.readthedocs.io/en/latest/additional_features.html?highlight=callable#extended-callable-types is accepted,
 # the '...' first argument of Callable can be replaced with VarArg(Tensor).
 # For now, we permit any input.
-# the '...' first argument of Callable can be replaced with VarArg(Tensor).
-# For now, we permit any input.
```
Comment (janeyx99) [inline torch/autograd/gradcheck.py:1431]: this was a small thing i noticed while reading through gradcheck--let me know if i should split this into another PR.

File: test/optim/test_swa_utils.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+    TestCase,
+    run_tests,
+    load_tests,
+    parametrize,
+    instantiate_parametrized_tests,
+)
+
+# load_tests from common_utils is used to automatically filter tests for
+# sharding on sandcastle. This line silences flake warnings
+load_tests = load_tests
```
Comment (janeyx99) [inline test/optim/test_swa_utils.py:18]: idk what this code is for, but it is now replicated across all three files

**Review by albanD — COMMENTED** [review]
Sounds good!

**Review by albanD — APPROVED** [review]
Sounds good!
